server/scsynth/scsynth_main.cpp

```diff
@@ -360,11 +351,49 @@ int main(int argc, char* argv[]) {
 
     EventLoop::run([world]() { World_WaitForQuit(world, true); });
 
+    return 0;
+}
+
 #ifdef _WIN32
+
+int wmain(int argc, wchar_t** wargv) {
+    // initialize winsock
+    WSAData wsaData;
+    int nCode;
+    if ((nCode = WSAStartup(MAKEWORD(1, 1), &wsaData)) != 0) {
+        scprintf("WSAStartup() failed with error code %d.\n", nCode);
+        return 1;
+    }
+
+    // convert args to utf-8
+    std::vector<char*> argv;
+    for (int i = 0; i < argc; i++) {
+        auto argSize = WideCharToMultiByte(CP_UTF8, 0, wargv[i], -1, nullptr, 0, nullptr, nullptr);
+        argv.push_back(new char[argSize]);
+        WideCharToMultiByte(CP_UTF8, 0, wargv[i], -1, argv[i], argSize, nullptr, nullptr);
+    }
+
+    // set codepage to UTF-8
+    static UINT gOldCodePage; // for remembering the old codepage when we switch to UTF-8
+    gOldCodePage = GetConsoleOutputCP();
```

Comment:
can combine to `auto oldCodePage = GetConsoleOutputCP();`. we only use the `g` prefix for globals